test/manager/baseManagerV2.spec.ts

```diff
@@ -0,0 +1,1710 @@
+    };
+    await setV2Setup.streamingFeeModule.initialize(setToken.address, streamingFeeSettings);
+
+    // Deploy BaseManager
+    baseManager = await deployer.manager.deployBaseManagerV2(
+      setToken.address,
+      operator.address,
+      methodologist.address,
+      [],
+      [[]]
+    );
+
+    // Transfer operatorship to BaseManager
+    await setToken.setManager(baseManager.address);
+
+    baseExtension = await deployer.mocks.deployBaseExtensionMock(baseManager.address);
+  });
+
+  addSnapshotBeforeRestoreAfterEach();
+
+  describe("#constructor", async () => {
+    let subjectSetToken: Address;
+    let subjectExtension: Address;
+    let subjectModule: Address;
+    let subjectAdditionalModule: Address;
+    let subjectAdditionalExtension: Address;
+    let subjectOperator: Address;
+    let subjectMethodologist: Address;
+    let subjectProtectedModules: Address[];
+    let subjectAuthorizedExtensions: Address[][];
+
+    beforeEach(async () => {
+      subjectSetToken = setToken.address;
+      subjectModule = setV2Setup.streamingFeeModule.address;
+      subjectAdditionalModule = setV2Setup.issuanceModule.address;
+      subjectExtension = baseExtension.address;
+      subjectAdditionalExtension = ADDRESS_ZERO; // Deploy as needed
+      subjectOperator = operator.address;
+      subjectMethodologist = methodologist.address;
+      subjectProtectedModules = [subjectModule];
+      subjectAuthorizedExtensions = [[]];
+    });
+
+    async function subject(): Promise<BaseManagerV2> {
+      return await deployer.manager.deployBaseManagerV2(
+        subjectSetToken,
+        subjectOperator,
+        subjectMethodologist,
+        subjectProtectedModules,
+        subjectAuthorizedExtensions
+      );
+    }
+
+    it("should set the correct SetToken address", async () => {
+      const retrievedICManager = await subject();
+
+      const actualToken = await retrievedICManager.setToken();
+      expect (actualToken).to.eq(subjectSetToken);
+    });
+
+    it("should set the correct Operator address", async () => {
+      const retrievedICManager = await subject();
+
+      const actualOperator = await retrievedICManager.operator();
+      expect (actualOperator).to.eq(subjectOperator);
+    });
+
+    it("should set the correct Methodologist address", async () => {
+      const retrievedICManager = await subject();
+
+      const actualMethodologist = await retrievedICManager.methodologist();
+      expect (actualMethodologist).to.eq(subjectMethodologist);
+    });
+
+    it("should not be initialized by default", async () => {
+      const retrievedICManager = await subject();
+
+      const initialized = await retrievedICManager.initialized();
+      expect(initialized).to.be.false;
+    });
+
+    describe("protectedModules: single, no extensions", () => {
+      beforeEach(() => {
+        subjectProtectedModules = [subjectModule];
+      });
+
+      // This test is borked... module initialized in before block..
```

Comment:
what's wrong with this test?

Reply:
Whoah! Glad you flagged this. The constructor *does not* add modules to the SetToken. This isn't possible - the `setToken.addModule` call reverts with `Manager only` I think because the manager contract doesn't technically exist yet.

So ... this situation is a little weird. You *could* protect a module in the Manager deployment which isn't on the token.

Have removed the bad test and replaced with one where non-added module is protected in the constructor and operator adds module after deployment. 

62c226a

Reply:
Yeah I think this is ok